### PR TITLE
[Sonic Frontiers] Add "Raise FPS Cap to 300 (experimental)" code [no CI]

### PR DESCRIPTION
**Pull request by NM-20**

### CODE REVIEW

File: HedgeModManager/Resources/Codesv2/SonicFrontiers.hmm [hunk 1]
```diff
@@ -744,4 +744,21 @@ Patch "Infinite Boost Gauge" by "Hyper"
...
+{
+    WriteAsmHook(
+        // 0x43960000 == 300.0f
+        @"
+            mov dword ptr [rcx+0x14],0x43960000
+            mov rbx,rcx
+            movss dword ptr [rcx+144h],xmm1
+        ",
+
+        // 0x15094C206
```
Comment (RagdollClash) [inline HedgeModManager/Resources/Codesv2/SonicFrontiers.hmm:759]:
```suggestion
        /* 0x15094C206 */
```

File: HedgeModManager/Resources/Codesv2/SonicFrontiers.hmm [hunk 2]
```diff
@@ -744,4 +744,21 @@ Patch "Infinite Boost Gauge" by "Hyper"
...
+
+Patch "Raise FPS Cap to 300 (experimental)" by "NM"
+{
+    WriteAsmHook(
+        // 0x43960000 == 300.0f
+        @"
+            mov dword ptr [rcx+0x14],0x43960000
+            mov rbx,rcx
+            movss dword ptr [rcx+144h],xmm1
+        ",
```
Comment (RagdollClash) [inline HedgeModManager/Resources/Codesv2/SonicFrontiers.hmm:757]:
```suggestion
    /* 0x43960000 == 300.0f */
    WriteAsmHook(
        @"
            mov dword ptr [rcx+0x14], 0x43960000
            mov rbx, rcx
            movss dword ptr [rcx+144h], xmm1
        ",
```

**Review by RagdollClash — CHANGES_REQUESTED** [review]
Is it possible to make this code uncap the framerate limit entirely instead? If not, then no worries.

Also the following change requests are for formatting consistency with other codes:

**Comment by NM-20** [comment]
> Is it possible to make this code uncap the framerate limit entirely instead? If not, then no worries.
> 
> Also the following change requests are for formatting consistency with other codes:

Best I could probably do with this opcode is set the cap to 999 or smth, but I might be able to pull off a complete removal. I'll give it a look in the morning.

**Comment by NM-20** [comment]
> > Is it possible to make this code uncap the framerate limit entirely instead? If not, then no worries.
> > Also the following change requests are for formatting consistency with other codes:
> 
> Best I could probably do with this opcode is set the cap to 999 or smth, but I might be able to pull off a complete removal. I'll give it a look in the morning.

Managed to kill the cap entirely, will submit a new PR in a bit. Btw, why does WriteAsmHook write 14 bytes for a JMP? With CE, only five bytes are written for a redirection

File: HedgeModManager/Resources/Codesv2/SonicFrontiers.hmm (hunk 2 above)
Comment (hyperbx) [reply HedgeModManager/Resources/Codesv2/SonicFrontiers.hmm:757]: That comment could be moved to an ASM comment instead, if you want to be explicit about what the value is;
```
mov dword ptr [rcx + 0x14], 0x43960000 ; 300.0f
```

**Comment by hyperbx** [comment]
> Btw, why does WriteAsmHook write 14 bytes for a JMP? With CE, only five bytes are written for a redirection

[CommonLoader uses a 14-byte jump by default for 64-bit processes](https://github.com/Sajidur78/CommonLoader/blob/master/CommonLoader/HookService.cpp#L76), but yeah, that should probably be determined by how far the hook address is away from the new code - that threw me off a couple times too.

**Pull request closed without merging**